inc/functions.php

```diff
@@ -334,14 +335,28 @@ function preferred_languages_load_textdomain_mofile( $mofile ) {
 
 	$current_locale = get_locale();
 
+	$first_mofile = null;
+
+	remove_filter( 'load_textdomain_mofile', 'preferred_languages_load_textdomain_mofile' );
+
 	foreach ( $preferred_locales as $locale ) {
 		$preferred_mofile = str_replace( $current_locale, $locale, $mofile );
 
 		if ( is_readable( $preferred_mofile ) ) {
-			return $preferred_mofile;
+			load_textdomain( $domain, $mofile );
```

Comment:
Should probably this from the looks of it:

```suggestion
			load_textdomain( $domain, $preferred_mofile );
```